url46a88cc932_toby1984_tinyc_tgz-pJ8-de_codesourcery_dcpu16_compiler_CompilerJ8/src/de/codesourcery/dcpu16/parser/Lexer.java

```diff
@@ -0,0 +1,433 @@
+                    return;
+                }
+                buffer.append( prefix );
+                continue;
+            }
+            
+            switch( c ) 
+            {
+                case ';':
+                    parseBuffer(bufferStartOffset);
+                    tokens.add( new Token(TokenType.SEMICOLON, offset, ";" ) );
+                    return;                    
+                case ',':
+                    parseBuffer(bufferStartOffset);
+                    tokens.add( new Token(TokenType.COMMA, offset, "," ) );
+                    return;                   
+                case '(':
+                    parseBuffer(bufferStartOffset);
+                    tokens.add( new Token(TokenType.PARENS_OPEN, offset,"(" ) );
+                    return;                
+                case ')':
+                    parseBuffer(bufferStartOffset);
+                    tokens.add( new Token(TokenType.PARENS_CLOSE , offset,")" ) );
+                    return;   
+                case '*':
+                    parseBuffer(bufferStartOffset);
+                    tokens.add( new Token(TokenType.STAR, offset,"*" ) );
+                    return;   
+                case '&':
+                    parseBuffer(bufferStartOffset);
+                    tokens.add( new Token(TokenType.AMPERSAND, offset,"&" ) );
+                    return;       
+                case '\\':
+                	parseBuffer(bufferStartOffset);
+                	tokens.add( new Token(TokenType.ESCAPE_CHARACTER, offset,"\\" ) );
+                	return;                   
+                case '[':
+                    parseBuffer(bufferStartOffset);
+                    tokens.add( new Token(TokenType.ANGLE_BRACKETS_OPEN, offset,"[" ) );
+                    return;                
+                case ']':
+                    parseBuffer(bufferStartOffset);
+                    tokens.add( new Token(TokenType.ANGLE_BRACKETS_CLOSE ,offset, "]" ) );
+                    return;
+                case '{':
+                    parseBuffer(bufferStartOffset);
+                    tokens.add( new Token(TokenType.BRACE_OPEN, offset,"{" ) );
+                    return;   
+                case '}':
+                    parseBuffer(bufferStartOffset);
+                    tokens.add( new Token(TokenType.BRACE_CLOSE ,offset, "}" ) );
+                    return;                        
+                case '\"':
+                    parseBuffer(bufferStartOffset);
+                    tokens.add( new Token(TokenType.DOUBLE_QUOTE, offset , "\"" ) );
+                    return;     
+                case '\'':
+                    parseBuffer(bufferStartOffset);
+                    tokens.add( new Token(TokenType.SINGLE_QUOTE, offset , "\'" ) );
+                    return;                      
+            }
+            buffer.append( c );
+        }
+        parseBuffer(bufferStartOffset);
+    }
+    
+    private void parseBuffer(int offset) 
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 321-383 into a separate method.